### PR TITLE
feat: error status persists until destroyed

### PR DESCRIPTION
**Pull request by dillonroller**

### CODE REVIEW

**Comment by dillonroller** [comment]
@aroller Ready to merge once its confirmed to be persisting the error status and successfully transitioning into `SHUTDOWN` on the simulator

File: src/super_lib/am_life_cycle_mediator.cpp [hunk 1]
```diff
@@ -101,7 +101,12 @@ bool AMLifeCycleMediator::setStatus(const LifeCycleStatus& status, LifeCycleInfo
   if (status < AMLifeCycleMediator::FIRST_STATUS || status > AMLifeCycleMediator::LAST_STATUS) 
     return false;
 
-  info.status = status;
+  //Error status should persist throughout flight regardless if we are now OK
+  if(info.status != LifeCycleStatus::ERROR)
+  {
+    info.status = status;
+  }
```
Comment (aroller) [inline src/super_lib/am_life_cycle_mediator.cpp:107]: Put a throttled warning, every few seconds, if in error and trying to set to ok or warning.

File: src/super_lib/am_life_cycle_mediator.cpp [hunk 2]
```diff
@@ -101,7 +101,12 @@ bool AMLifeCycleMediator::setStatus(const LifeCycleStatus& status, LifeCycleInfo
   if (status < AMLifeCycleMediator::FIRST_STATUS || status > AMLifeCycleMediator::LAST_STATUS) 
     return false;
 
-  info.status = status;
+  //Error status should persist throughout flight regardless if we are now OK
```
Comment (aroller) [inline src/super_lib/am_life_cycle_mediator.cpp]: add conditional check only to continue this block if the given status is different than the existing.

**Comment by dillonroller** [comment]
@aroller Passes happy case on simulator.

File: src/super_lib/am_life_cycle.cpp [hunk 3]
```diff
@@ -336,10 +336,16 @@ LifeCycleStatus AMLifeCycle::getStatus() const
 
 bool AMLifeCycle::setStatus(const LifeCycleStatus status)
 {
-  if (life_cycle_mediator_.setStatus(status, life_cycle_info_))
+  if(life_cycle_info_.status == LifeCycleStatus::ERROR && status != LifeCycleStatus::ERROR)
```
Comment (aroller) [inline src/super_lib/am_life_cycle.cpp:339]: clean and simple.

**Comment by aroller** [comment]
Tested failure on gengar  
image

**Comment by aroller** [comment]
tested on gengar to keep the shutdown state.

**Pull request merged**